src/coreComponents/mesh/generators/VTKMeshGenerator.cpp

```diff
@@ -0,0 +1,711 @@
+      for( localIndex v = 0; v < numPointsPerCell; v++)
+      {
+        cellToVertex[cellCount][v] = currentCell->GetPointId(v);
+      }
+      localToGlobal[cellCount] = globalCellIdsDataArray.GetValue(c);
+      cellCount++;
+    }
+  }
+}
+
+
+/**
+ * @brief Write all the cell blocks
+ * @param[in] domain the domain in which the cell blocks will be written
+ * @param[in] regionsHex map from region index to the hexahedron indexes in this region
+ * @param[in] regionsTetra map from region index to the tetra indexes in this region
+ * @param[in] regionsWedges map from region index to the wedges indexes in this region
+ * @param[in] regionsPyramids map from region index to the pyramids indexes in this region
+ * @param[in] regions a set containing all the region indexes
+ * @param[in] arraysToBeImported a vector containing all the vtkDataArray that can be imported
+ * @param[in] mesh the vtkUnstructuredGrid that is loaded
+ */
+void writeCellBlocks( DomainPartition& domain,
+                      std::map<int,std::vector<vtkIdType >> & regionsHex, std::map<int,std::vector<vtkIdType >> & regionsTetra,
+                      std::map<int,std::vector<vtkIdType >> & regionsWedges, std::map<int,std::vector<vtkIdType >> & regionsPyramids,
+                      std::set< int > & regions, std::vector< vtkDataArray * > & arraysToBeImported,
+                      vtkUnstructuredGrid & mesh)
+{
+  CellBlockManager & cellBlockManager = domain.getGroup< CellBlockManager >( keys::cellManager );
+
+  if( !regions.empty() )
+  {
+    for (int region : regions)
+    {
+      std::map< int, std::vector<vtkIdType >>::iterator it;
+      it = regionsHex.find(region);
+      if( it != regionsHex.end())
+      {
+        writeCellBlock("hexahedron", it->second, it->first, VTK_HEXAHEDRON, cellBlockManager, arraysToBeImported, mesh);
+      }
+      it = regionsTetra.find(region);
+      if( it != regionsTetra.end())
+      {
+        writeCellBlock("tetrahedron", it->second, it->first, VTK_TETRA, cellBlockManager, arraysToBeImported, mesh);
+      }
+      it = regionsWedges.find(region);
+      if( it != regionsWedges.end())
+      {
+        writeCellBlock("wedges", it->second, it->first, VTK_WEDGE, cellBlockManager, arraysToBeImported,mesh);
+      }
+      it = regionsPyramids.find(region);
+      if( it != regionsPyramids.end())
+        writeCellBlock("pyramids", it->second, it->first, VTK_PYRAMID, cellBlockManager, arraysToBeImported, mesh);
+    }
+  }
+  else
+  {
+      writeCellBlock("hexahedron", regionsHex[-1], -1, VTK_HEXAHEDRON, cellBlockManager, arraysToBeImported, mesh);
+      writeCellBlock("tetrahedron", regionsTetra[-1], -1, VTK_TETRA, cellBlockManager, arraysToBeImported, mesh);
+      writeCellBlock("wedges", regionsWedges[-1], -1, VTK_WEDGE, cellBlockManager, arraysToBeImported, mesh);
+      writeCellBlock("pyramids", regionsPyramids[-1], -1, VTK_PYRAMID, cellBlockManager, arraysToBeImported, mesh);
```

Comment:
Decide on singular vs plural names for cell blocks? Doesn't matter which one, as long as it's consistent.